### DISCUSSION TITLE
EC2 + ALB + ACM + Route 53 - Outline Deployment

### DISCUSSION THREAD

**Discussion by KarthikR511** [outline/outline#7806] [Self Hosting]
### Discussed in https://github.com/outline/outline/discussions/7803

<div type='discussions-op-text'>

<sup>Originally posted by **KarthikR511** October 19, 2024</sup>
Hello,

### Environment
- Deployment: EC2 instance
- Database: PostgreSQL
- Cache: Redis
- SSL Termination: Application Load Balancer (ALB)
- SSL Certificate: AWS Certificate Manager (ACM)
- DNS: Route 53

### Configuration
relevant `.env` settings:
```
URL=https://domain.com
TRUST_PROXY=true
FORCE_HTTPS=false
```

### Current Status
- Health check endpoint is responding OK
- `curl` requests to the server are successful
- ALB target group shows the instance as healthy

### Problem
When attempting to access the Outline home page, I'm encountering an error. The browser console shows:

![Error Screenshot](https://github.com/user-attachments/assets/e29ef9ff-568d-470e-8ec3-4f74cc44950a)

The error suggests issues with loading `auth.info` and `auth.config`(Found through Browser --> Inspect), resulting in 401 and 403 errors respectively.

### Troubleshooting Steps Taken
1. Verified ALB configuration and target group health
2. Confirmed Route 53 settings pointing to ALB
3. Checked EC2 security group settings
4. Reviewed Outline `.env` configuration

### Questions
1. Are there any additional configuration steps needed for Outline to work correctly behind an ALB?
2. Could this be related to how Outline is handling the SSL termination at the ALB?
3. Are there any logs I should check or provide to help diagnose this issue?

Any guidance on resolving this or further troubleshooting steps would be greatly appreciated.</div>

**Comment by linear** [comment]
<p><a href="https://linear.app/outline/issue/OLN-711/ec2-alb-acm-route-53-outline-deployment">OLN-711 EC2 + ALB + ACM + Route 53 - Outline Deployment</a></p>
